core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -983,17 +985,16 @@ protected R loadBuild(File dir) throws IOException {
      * @see TransientProjectActionFactory
      */
     @SuppressWarnings("deprecation")
+    @Nonnull
     @Override
     public List<Action> getActions() {
         // add all the transient actions, too
-        List<Action> actions = new Vector<Action>(super.getActions());
+        List<Action> actions = new Vector<>(getPersistedActions());
```

Comment:
ArrayList now?

Reply:
Yes, ArrayList

Reply:
@jglick wdyt?

Reply:
Vector has built-in concurrency support while ArrayList does not offer one. I am not sure whether there is a reliance on that anywhere in Jenkins, but my recommendation would be to create a follow-up ticket for that